source/Patches/CrewmateRoles/ProphetMod/Revelation.cs

```diff
@@ -1,44 +1,35 @@
-using System;
+using System.Linq;
 using HarmonyLib;
 using TownOfUs.Roles;
 
 namespace TownOfUs.CrewmateRoles.ProphetMod
 {
-    [HarmonyPatch(typeof(PlayerControl), nameof(PlayerControl.FixedUpdate))]
+    [HarmonyPatch(typeof(PlayerControl), nameof(PlayerControl.CompleteTask))]
     public class Revelation
     {
         public static void Postfix(PlayerControl __instance)
         {
-            if (
-                PlayerControl.AllPlayerControls.Count <= 1
-                || PlayerControl.LocalPlayer == null
-                || PlayerControl.LocalPlayer.Data == null
-                || MeetingHud.Instance
-                || !PlayerControl.LocalPlayer.CanMove
-                || !PlayerControl.LocalPlayer.Is(RoleEnum.Prophet)
-                || PlayerControl.LocalPlayer.Data.IsDead
-                )
+            if (!PlayerControl.LocalPlayer.Data.IsDead
+                && PlayerControl.LocalPlayer.Is(RoleEnum.Prophet)
+                && ShouldReveal(__instance))
             {
-                return;
+                Role.GetRole<Prophet>(PlayerControl.LocalPlayer).Revelation();
             }
-
-            if (!IsTimeForRevelation())
-            {
-                return;
-            }
-
-            Role.GetRole<Prophet>(PlayerControl.LocalPlayer).Revelation();
         }
 
-        private static bool IsTimeForRevelation()
+        private static bool ShouldReveal(PlayerControl __instance)
         {
-            var now = DateTime.UtcNow;
+            var taskInfos = __instance.Data.Tasks.ToArray();
+            var allTasksCount = taskInfos.Count;
+            var maxRevealsCount = 
+                CustomGameOptions.ProphetTotalReveals < 1 ? 1 : CustomGameOptions.ProphetTotalReveals;
```

Comment:
Isn't `ProphetTotalReveals` bounded to be a minimum of 1?

Reply:
just to guard against any bugs in CustomNumberOption, but maybe too defensive.

Reply:
being too defensive. rewritten to remove divison eliminating both risk of float imprecision and divide-by-zero.